lib/index.js

```diff
@@ -10,12 +14,60 @@ var async = require('async');
  * @param {object} changes object provided by anyFetch's API. Update this object to send document's modification to anyFetch's API.
  * @param {function} cb Callback, first parameter is the error if any, then the processed data
  */
-module.exports = function(path, document, changes, cb) {
-  async.waterfall([
-    function(cb) {
-      cb();
+
+ module.exports = function(path, document, changes, finalCallback) {
+ 	async.waterfall([
+ 		function(cb) {
+ 			var outfile = crypto.randomBytes(20).toString('hex');
+ 			var options = [
+        '-density 200', // To sharpen the image
+        ];
+        shellExec('convert ' + path + ' ' + '/tmp/' + outfile + '.png', function(err, stdout, stderr) {
+        	if(err) {
+        		return cb(new HydrationError([err, stderr, stdout]));
+        	} else {
+        		var outputFiles = [];
+        		fs.readdir('/tmp/', function(err, res){
+        			for (var i = 0; i < res.length; i++){
```

Comment:
try not to use the for() loop, instead use the functional equivalent (forEach, map, reduce, filter)